nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQueryStreaming.java

```diff
@@ -191,6 +195,16 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                     lmapr.add(convertMapRecord(((MapRecord) mapr).toMap()));
                 }
                 result.put(key, lmapr);
+            } else if (obj instanceof Timestamp) {
+                LocalDateTime dateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(((Timestamp) obj).getTime()), ZoneOffset.UTC);
+                DateTimeFormatter timestampFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSSSSS");
+                result.put(key, dateTime.format(timestampFormatter));
```

Comment:
Similar to the `java.sql.Date` and `java.sql.Time` fields, `java.sql.Timestamp` does not necessarily indicate a particular time zone. In this case, the [Datetime type](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#datetime_type) seems to be a closer approximation of `java.sql.Timestamp`. Converting `java.sql.Timestamp` to `java.time.LocalDateTime` would allow using the `toString()` method for formatting while also avoiding the time zone conversion.

```suggestion
                final LocalDateTime localDateTime = ((Timestamp) obj).toLocalDateTime();
                result.put(key, localDateTime.toString());
```

Reply:
Same reason as with Time, it is timezone dependent and comes out 2021-07-18 14:35:24.